wandb/sdk/launch/agent/agent.py

```diff
@@ -121,25 +126,62 @@ def finish_job_id(self, job_id: Union[str, int]) -> None:
     @staticmethod
     def _parse_runner_config(config: Dict[str, Any]) -> Dict[str, Any]:
         """
-            Parses a RunnersConfig block from the agent config. Returns a dictionary
-            of support runners in the form of:
-            {
-                kubernetes: {
-                    namespace: "wandb"
-                },
-                sagemaker: {
-
-                },
-                local-process: {
-                    labels: [ "gpu-pool" ]
-                },
-                local-container: {
-                    labels: [ "gpu-pool"]
-                }
+        Parses a RunnersConfig block from the agent config. Returns a dictionary
+        of support runners in the form of:
+        {
+            kubernetes: {
+                namespace: "wandb"
+            },
+            sagemaker: {
+
+            },
+            local-process: {
+                labels: [ "gpu-pool" ]
+            },
+            local-container: {
+                labels: [ "gpu-pool"]
             }
+        }
         """
         pass
 
+    def _get_valid_resource_configs(self, entity: str) -> list[RunQueue]:
+        """
+        Given an entity, return a list of run queues associated with that entity.
+        """
+        QUERY = gql(
+            """
+            query Query {
+                runQueues(entityName: String!): [RunQueue!]
+            }
+            """
+        )
+
+        run_queues = self._api.client.execute(QUERY, {"entityName": entity})
+        resource_configs = [json.loads(q["resourceConfig"]) for q in run_queues]
+        valid_resource_configs = [
+            rc for rc in resource_configs if rc["runner"] in self._configured_runners
+        ]
+        return valid_resource_configs
```

Comment:
can we also support label match for local process and local container runners?

Reply:
yep